examples/AspNetCore/Controllers/WeatherForecastController.cs

```diff
@@ -24,16 +26,23 @@ public class WeatherForecastController : ControllerBase
 {
     private static readonly string[] Summaries = new[]
     {
-            "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
     };
 
     private static readonly HttpClient HttpClient = new();
 
     private readonly ILogger<WeatherForecastController> logger;
+    private readonly ActivitySource activitySource;
+    private readonly Counter<int> counter;
 
-    public WeatherForecastController(ILogger<WeatherForecastController> logger)
+    public WeatherForecastController(ILogger<WeatherForecastController> logger, ActivitySource activitySource, Meter meter)
```

Comment:
@dnelson-relativity In the current shape of the .NET API I think it is an anti-pattern to use DI for ActivitySource & Meter. They are really intended to be used statically. If you really want to inject them, might be better to create a telemetry helper class which exposes source + meter for the app logic and can be safely injected.

/cc @noahfalk

Reply:
Happy to change this to static but I am curious if you could provide more context as to the reason why. I think the DI pattern is a bit more natural for an AspNetCore developer so that is what I chose. Perhaps the examples/docs should speak to that reasoning?

In this case is there something fundamentally different between using it as a singleton and using it statically?

This example is pretty simple so I can just new up the `ActivitySource`/`Meter` in `WeatherForecastController` but I wouldn't want to make people think they should be doing that in every controller. Do you think that is ok or should I create these elsewhere?

Reply:
> In this case is there something fundamentally different between using it as a singleton and using it statically?

The main issue conceptually is it is perfectly acceptable and expected there will many `ActivitySource`s and many `Meter`s in a process so who wins when multiple things call `AddSingleton`? 😄 [Here is what ASP.NET Core does](https://github.com/dotnet/aspnetcore/blob/ea2c82c0f7b0af55dacaea9f37fbe6d5e176f9db/src/Hosting/Hosting/src/WebHostBuilder.cs#L288) so this code registering its own singleton is probably taking over ASP.NET Core's and causing some very odd telemetry!

> This example is pretty simple so I can just new up the ActivitySource/Meter in WeatherForecastController but I wouldn't want to make people think they should be doing that in every controller. Do you think that is ok or should I create these elsewhere?

Probably best to keep it simple and just new them up in the controller. Best practice would probably be build something which plugs into MVC like an action filter (are those still a thing?) but that might be more code than it is worth for an example. It is very hard to show something useful without too much code in these docs/examples! If you are super passionate about it, we could have a more advanced example. Not sure how everyone would feel about that, but I'm not opposed to it.

Reply:
I see what you mean. If I update the `TracerProvider` configuration to the following and disable the auto instrumentation
```
builder
    .AddSource("Microsoft.AspNetCore", "Examples.AspNetCore")
    .SetSampler(new AlwaysOnSampler());
    //.AddAspNetCoreInstrumentation() -> Disabled
```
I get this unexpected result
![image](https://user-images.githubusercontent.com/55757989/216161247-68df1015-6c7e-442a-97d7-2cba5f9ac17f.png)
Interestingly this doesn't happen with the auto instrumentation which is why I never encountered this before.

I updated the example to use statics. On the fence about the "Examples.AspNetCore" magic string but it removes the need for some global constant or passing of additional configuration to the controller. Part of me thinks it is more explicit with the duplication.

Reply:
I don't know that it is written down anywhere, but we had past discussions with @davidfowl on this topic and agreed that:

- It was a mistake for ASP.NET to directly inject an instance of ActivitySource, but given back-compat concerns for now we are keeping the mistake as-is rather than changing it.
- In general developers shouldn't add types to a DI container if their code doesn't define the type. If you want to access an instance of some common type, a better pattern is to define a type that will contain it and inject the containing type. For example instead of injecting ActivitySource, define `class MyLibraryTelemetry { ActivitySource Source; }` and inject MyLibraryTelemetry instead. This avoids the type collisions with other components that might also like to locate an ActivitySource via the DI container.